escape_roomba/thread_manager.py

```diff
@@ -77,261 +81,449 @@ async def _on_ready(self):
+            remaining = self._CHANNEL_LENGTH - len(mash) - len(to_add)
             if remaining >= 0:
-                first_words += f'{dash}{word}'
-                continue
-            if len(first_words) < 10:
-                first_words += f'{dash}{word[:remaining]}'
-            first_words += self._ELLIPSIS
+                mash += to_add
+                continue  # The word fits in its entirety; keep going.
+            if len(mash) < self._CHANNEL_LENGTH // 2:
+                # Chop long words if needed to get a reasonable channel mash.
+                mash += f'{to_add[:remaining]}'
+            mash += self._ELLIPSIS
             break
 
-        ci, mi = message.channel.id, message.id
-        name = f'{self._THREAD_EMOJI}{first_words}'
-        topic = f'[id={ci:x}/{mi:x}]'
-        if self._logger.isEnabledFor(logging.DEBUG):
-            self._logger.debug(f'Creating #{name}:\n'
-                               f'    oc={fid(ci)} om={fid(mi)}\n'
-                               f'    topic="{topic}"')
+        basic_name = f'{self._THREAD_EMOJI}{mash or "thread"}'
+        name, number = basic_name, 1  # Add a suffix if needed for uniqueness.
+        existing = set(c.name for c in origin_message.guild.channels)
+        while name in existing:
+            number += 1
+            name = f'{basic_name}-{number}'
 
-        channel = await message.guild.create_text_channel(
-            name=name, category=message.channel.category,
-            position=len(message.guild.channels), topic=topic,
+        # Special name and topic format enables recognizability.
+        tag = f' started by <@{creating_user.id}>' if creating_user else ''
+        topic = (f'Thread{tag} for [<#{ci}>/{mi:x}].')
+        if self._logger.isEnabledFor(logging.DEBUG):
+            self._logger.info('Creating channel:\n'
+                              f'    "{origin_message.guild.name}" #{name}\n'
+                              f'    topic "{topic}"\n'
+                              f'    origin {fobj(m=origin_message)}')
+
+        channel = await origin_message.guild.create_text_channel(
+            name=name, category=origin_message.channel.category,
+            position=len(origin_message.guild.channels), topic=topic,
             reason='Thread creation')
 
-        t = self._Thread(origin_channel_id=ci, origin_message_id=mi,
-                         thread_channel=channel)
-        self._thread_by_origin.setdefault(ci, {})[mi] = t
-        self._thread_by_channel[channel.id] = t
+        thread = self._Thread(channel=channel, origin_cid=ci, origin_mid=mi)
+        thread.intro_messages = []  # New channel is empty!
+        self._thread_by_origin.setdefault(ci, {})[mi] = thread
+        self._thread_by_channel[channel.id] = thread
+        return thread
 
-    async def _async_message_gone(self, channel_id, message_id):
-        """Handles a message that was deleted."""
+    async def _async_delete_locked_thread(self, thread):
```

Comment:
Thinking more about this, I'm wondering if it would be better to archive thread channels in some way, if they have content, instead of entirely deleting them. (Assuming that Discord does not have some unreasonably low limit on number of archived channels.) The intent would be to keep a record of the conversation that happened, and be able to refer to it later.

Reply:
There is, of course, the issue of visibility. If everyone removes their 🧵  reacji, and thus nobody has visibility of the thread, maybe it's entirely moot to archive it? But I feel like preserving the conversations and making them available in some archival way might be useful.